bin/find-never-defined-css-variables/find-never-defined-css-variables.unit.spec.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  extractVariableDefinitionsFromFileContent,
+  extractVariableUsagesFromFileContent,
+} from "./find-never-defined-css-variables";
+
+describe("extractVariableDefinitionsFromFileContent", () => {
+  it("should extract standard CSS variable definitions", () => {
+    const content = `
+      .my-class {
+        --color-brand: #509ee3;
+        --color-text: #000;
+      }
+    `;
+    const result = extractVariableDefinitionsFromFileContent(content);
+    expect(result).toEqual(new Set(["--color-brand", "--color-text"]));
+  });
+
+  it("should extract CSS-in-JS variable definitions", () => {
+    const content = `
+      const MyContainer1 = styled.div\`
+        --theme-primary: "blue",
+      \`;
+
+      const MyContainer2 = styled.div({
+        --theme-secondary: "#fff",
+      })
+    `;
+    const result = extractVariableDefinitionsFromFileContent(content);
+    expect(result).toEqual(new Set(["--theme-primary", "--theme-secondary"]));
+  });
+
+  it("should handle mixed quotes in definitions", () => {
+    const content = `
+      const theme = {
+        '--single-quote': 'value',
+        "--double-quote": "value",
+        \`--backtick\`: \`value\`,
+      };
+    `;
+    const result = extractVariableDefinitionsFromFileContent(content);
+    expect(result).toEqual(
+      new Set(["--single-quote", "--double-quote", "--backtick"]),
+    );
+  });
+});
+
+describe("extractVariableUsagesFromFileContent", () => {
+  it("should extract usage from css syntax", () => {
+    const content = `
+      .button {
+        color: var(--color-brand);
+        background: var(--color-background);
+      }
+    `;
+    const result = extractVariableUsagesFromFileContent(content);
+    expect(result).toEqual(new Set(["--color-brand", "--color-background"]));
+  });
+
+  it("should extract usage from css-in-js syntax", () => {
+    const content = `
+      const MyDiv1 = styled.div\`
+        color: var(--theme-color);
+      \`;
+
+      const MyDiv2 = styled.div({
+        background: var(--theme-background);
+      })
+    `;
+    const result = extractVariableUsagesFromFileContent(content);
+    expect(result).toEqual(new Set(["--theme-color", "--theme-background"]));
```

Comment:
you you add tests that contains both css declaration and css usage, just to ensure these functions extract the expected parts? i.e. add a test for each function that contains both CSS custom property definition and CSS custom property usage

Reply:
Good point! like this? 9b20b2b03466fcc5351475487f31c418ed8e12df

Reply:
Exactly, thanks you!